### PR TITLE
Fix API plugins error handling - read PR description

### PR DESCRIPTION
**Pull request by Omri-Levy**
Need to ensure 3rd party processes plugins transition to success before merging.

- fix(workflow-core): fixed error handling for api plugins
- style(*): format changes

## Summary by CodeRabbit

- **New Features**
	- Updated dependency versions in multiple projects to enhance functionality and compatibility.
- **Bug Fixes**
	- Improved error handling and reporting in the workflow-runner tests.
- **Documentation**
	- Updated changelogs across various packages to reflect new version entries and dependency updates.
- **Chores**
	- Incremented version numbers in relevant package.json files to indicate minor releases.

### CODE REVIEW

File: apps/backoffice-v2/src/common/components/molecules/ProcessTracker/hooks/useProcessTracker/process-tracker-adapters/merchant-monitoring.process-tracker.ts [hunk 1]
```diff
@@ -37,8 +37,8 @@ export class MerchantMonitoringProcessTracker implements IProcessTracker {
   }
 
   private getIconKeyByState(tags: string[]): JSX.Element {
-    return tagToIcon[tags[0] as keyof typeof tagToIcon]
-      ? tagToIcon[tags[0] as keyof typeof tagToIcon]
-      : Icon.INDICATOR;
+    const tag = tags?.find(tag => tagToIcon[tag as keyof typeof tagToIcon]);
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/common/components/molecules/ProcessTracker/hooks/useProcessTracker/process-tracker-adapters/merchant-monitoring.process-tracker.ts:40]: Parity with other places that do the same thing.

File: apps/backoffice-v2/src/common/components/molecules/ProcessTracker/hooks/useProcessTracker/process-tracker-adapters/third-party.proces-tracker.tsx [hunk 2]
```diff
@@ -20,7 +20,8 @@ export class ThirdPartyProcessTracker implements IProcessTracker {
       this.plugins
         ?.filter(({ name }) => pluginsWhiteList.includes(name as (typeof pluginsWhiteList)[number]))
         ?.map(({ displayName, name }) => {
-          const pluginStatus = this.getPluginByName(name)?.status ?? ProcessStatus.DEFAULT;
+          const plugin = this.getPluginByName(name);
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/common/components/molecules/ProcessTracker/hooks/useProcessTracker/process-tracker-adapters/third-party.proces-tracker.tsx:23]: This change made it easier for me to debug the plugin separately from its status.

File: packages/workflow-core/src/lib/workflow-runner.ts [hunk 3]
```diff
@@ -744,11 +744,22 @@ export class WorkflowRunner {
...
+        this.context,
+        responseBody,
+        `pluginsOutput.${apiPlugin.name}`,
+      );
+    }
+
+    if (error) {
+      this.context = this.mergeToContext(
+        this.context,
+        { name: apiPlugin.name, error, status: ProcessStatus.ERROR },
```
Comment (Omri-Levy) [inline packages/workflow-core/src/lib/workflow-runner.ts:760]: Main changes are `name` and `status`.

Comment (alonp99) [reply packages/workflow-core/src/lib/workflow-runner.ts:760]: @Omri-Levy didn't we said we are going to remove the dependency on `plugin.name` and use the key instead?

Comment (alonp99) [reply packages/workflow-core/src/lib/workflow-runner.ts:760]: or its part of the second phase

**Review by alonp99 — APPROVED** [review]

**Pull request merged**